src/asn1/scheme/strings.rs

```diff
@@ -68,15 +68,24 @@ pub struct BitStringNodeProps {
 
 #[function_component(BitStringNode)]
 pub fn bit_string(props: &BitStringNodeProps) -> Html {
-    let bits = props.node.raw_bits()[1..]
-        .iter()
-        .map(|byte| format!("{:08b}", byte))
-        .fold(String::new(), |mut ac, new| {
-            ac.push_str(&new);
-            ac
-        });
+    let raw_bits = props.node.raw_bits();
     let bits_amount = props.node.bits_amount();
-    let bits = &bits[0..bits_amount];
+
+    let bits = if raw_bits.len() > 1 {
+        let mut bits = String::with_capacity((raw_bits.len() - 1) * 8);
+        for byte in &raw_bits[1..] {
+            write!(bits, "{:08b}", byte).unwrap();
+        }
+        bits
+    } else {
+        String::new()
+    };
+
+    let display_bits = if bits.len() >= bits_amount {
+        &bits[0..bits_amount]
+    } else {
+        &bits[..]
+    };
```

Comment:
this code can be simplified:

```suggestion
    let display_bits = &bits[0..bits_amount.min(bits.len())];
```